sonic_platform_base/watchdog_base.py

```diff
@@ -0,0 +1,50 @@
+#
+# watchdog_base.py
+#
+# Abstract base class for implementing a platform-specific class with which
+# to interact with a hardware watchdog module in SONiC
```

Comment:
how do we know if the platform has such capability or not? do we have query API for this?

Reply:
To avoid the need for additional functions with which to query device/capability, the current design is such that the querying is done by getting the device object. If the object is null (None), the device/capability is not present on the platform. For example:

```
#!/usr/bin/env python

import sonic_platform

platform = sonic_platform.platform.Platform()

chassis = platform.get_chassis()
if not chassis:
    print("Error getting chassis!")
    sys.exit(1)

watchdog = chassis.get_watchdog()
if not watchdog:
    print("Watchdog not present!")
else:
    watchdog.arm(10)
```